src/yo/yo.ts

```diff
@@ -92,57 +82,16 @@ export default class Yeoman {
 			generator = generator.slice(prefix.length);
 		}
 
-		return new Promise((resolve, reject) => {
-			Promise.resolve(window.showQuickPick(new Promise((res, rej) => {
-				setImmediate(() => {
-					try {
-						this._env.run(generator, this.done)
-							.on('npmInstall', () => {
-								this.setState('install node dependencies');
-							})
-							.on('bowerInstall', () => {
-								this.setState('install bower dependencies');
-							})
-							.on('error', err => {
-								if (!(err instanceof EscapeException)) {
-									window.showErrorMessage(err.message);
-									throw err;
-								}
-							})
-							.on('end', () => {
-								this.clearState();
-								console.log(`${EOL}${figures.tick} done`);
-							});
-
-						resolve();
-					} catch (err) {
-						reject(err);
-					}
-
-					rej();
-				});
-			}))).catch(err => {
-				// do nothing because the input is always rejected
-			});
+		return this._env.run(generator, this.done).then(() => {
```

Comment:
Do we no longer need the bower and node dependencies installed here? Or are they installed when the extension itself is added without additional help?

Reply:
yeoman generator can install bower, npm, yarn, etc. But yeoman environemnt package doesn't expose these events on the run method. Only promise to the end of the generation process is returned.